src/site/xdoc/server/dev-extend.xml

```diff
@@ -72,6 +72,60 @@
   
   </subsection>
 
+    <subsection name="Extension and Guice wiring">
+
+        <p>This section do not concern the Spring / server/app wiring.</p>
+
+        <p>Guice applications let you load several type of user defined components. These components includes:</p>
+
+        <ul>
+            <li><a href="https://github.com/apache/james-project/blob/master/mailet/api/src/main/java/org/apache/mailet/Mailet.java">Mailets</a></li>
+            <li><a href="https://github.com/apache/james-project/blob/master/mailet/api/src/main/java/org/apache/mailet/Matcher.java">Matchers</a></li>
+            <li><a href="https://github.com/apache/james-project/blob/master/mailbox/api/src/main/java/org/apache/james/mailbox/events/MailboxListener.java">Mailbox Listeners</a></li>
+            <li><a href="https://github.com/apache/james-project/blob/master/mailbox/api/src/main/java/org/apache/james/mailbox/extension/PreDeletionHook.java">PreDeletion hooks</a></li>
+            <li><a href="https://github.com/apache/james-project/blob/master/protocols/api/src/main/java/org/apache/james/protocols/api/handler/ProtocolHandler.java">Protocol handlers</a> (SMTP/LMTP/POP3)</li>
+            <li><a href="https://github.com/apache/james-project/blob/master/server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/Routes.java">Additional webAdmin routes</a></li>
+        </ul>
+
+        <p>Given a custom maven module implementing one or more of the following APIs, first create a <b>jar-with-dependencies</b>
+        of your module. Then copy the jar-with-dependencies into the <code>./extensions-jars</code> folder of your guice
+        James server installation.</p>
+
+        <p>Note: Transitive dependency resolution is only done if you are packaging your extension into a jar-with-dependencies.
+        Your extension class-loader will be the modified one (James and extensions-jars content) but James class-loader
+        stays unmodified everywhere else.</p>
```

Comment:
maybe you can give a link to a maven or java documentation explaining what is a jar-with-dependencies?